Extension/src/LanguageServer/customProviders.ts

```diff
@@ -158,7 +158,7 @@ export class CustomConfigurationProviderCollection {
     }
 
     public add(provider: CustomConfigurationProvider, version: Version): boolean {
-        if (new CppSettings(ext.getActiveClient().RootUri).intelliSenseEngine === "Disabled") {
+        if (new CppSettings(ext.getActiveClient().RootUri).intelliSenseEngine === "disabled") {
```

Comment:
Doesn't this need to be compared against in a case insensitive way?  Perhaps there is a `toLowerCase()` needed here?

Reply:
Alternatively, you could lower case these in the getters in `settings.ts` so they always return lower case strings.  Then all comparisons between settings originating from these getters would not need any conversions.  That would seem simpler, as we'd need to deal with potentially different casing in much fewer places (no other places?).

Also, for setters that are simply enabled/disabled, it would be most efficient if they were converted into bools by the getters in `settings.ts`, so could thereafter be evaluated as bools instead of needing to do any more expensive string conversions/comparisons.